python/lsst/ts/standardscripts/base_offset_tcs.py

```diff
@@ -0,0 +1,200 @@
+                properties:
+                    x:
+                        description: Offset in camera x-axis (arcsec).
+                        type: number
+                    y:
+                        description: Offset in camera y-axis (arcsec).
+                        type: number
+                required: ["x","y"]
+              offset_rot:
+                type: object
+                description: Offset rotator angle.
+                properties:
+                    rot:
+                        description: Offset rotataor (degrees).
+                        type: number
+                required: ["rot"]
+              reset_offsets:
+                type: object
+                description: Reset offsets
+                properties:
+                    reset_absorbed:
+                        description: Reset absorbed offset? If unsure, set True
+                        type: boolean
+                    reset_non_absorbed:
+                        description: Reset non-absorbed offset? If unsure, set True
+                        type: boolean
+                required: ["reset_absorbed","reset_non_absorbed"]
+              relative:
+                description: If `True` (default) offset is applied relative to the current
+                    position, if `False` offset replaces any existing offsets.
+                type: boolean
+                default: True
+              absorb:
+                description: If `True`, offset should be absorbed and persisted between
+                    slews.
+                type: boolean
+                default: False
+            additionalProperties: false
+            oneOf:
+                - required: ["offset_azel"]
+                - required: ["offset_radec"]
+                - required: ["offset_xy"]
+                - required: ["offset_rot"]
+                - required: ["reset_offsets"]
+        """
+        return yaml.safe_load(schema_yaml)
+
+    async def configure(self, config):
+        """Configure script.
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Script configuration, as defined by `schema`.
+        """
+
+        self.offset_azel = getattr(config, "offset_azel", None)
+        self.offset_radec = getattr(config, "offset_radec", None)
+        self.offset_xy = getattr(config, "offset_xy", None)
+        self.offset_rot = getattr(config, "offset_rot", None)
+        self.reset_offsets = getattr(config, "reset_offsets", None)
+
+        self.relative = config.relative
+        self.absorb = config.absorb
+
+    def set_metadata(self, metadata):
+        metadata.duration = 10
+
+    async def assert_feasibility(self) -> None:
+        """Verify that the telescope and is in a feasible state to
+        execute the script.
+        """
+
+        await self.tcs.assert_all_enabled()
+
+    async def run(self):
+        await self.assert_feasibility()
+
+        if self.offset_azel:
```

Comment:
I think it is better if you check that this is not None. Doing it this way may give the impression that `self.offset_azel` is a boolean. Also, I think it is not safe to assume that `None==False`.

Consider changing these to:

```
if self.offset_azel is not None:
    ...
```